services/insights-handler/internal/handler/main.go

```diff
@@ -0,0 +1,722 @@
+
+		objectName := strings.ToLower(fmt.Sprintf("%s-%s-%s-%s.json", insights.InputType, resource.Project, resource.Environment, resource.Service))
+		reader := bytes.NewReader(b)
+		info, putObjErr := minioClient.PutObject(ctx, h.S3Config.Bucket, objectName, reader, reader.Size(), minio.PutObjectOptions{})
+		if putObjErr != nil {
+			return putObjErr
+		}
+
+		log.Printf("Successfully uploaded %s of size %d\n", objectName, info.Size)
+	}
+
+	if incoming.BinaryPayload != nil {
+		for _, p := range incoming.BinaryPayload {
+			result, err := decodeString(p)
+			if err != nil {
+				fmt.Errorf(err.Error())
+			}
+			resultJson, _ := json.MarshalIndent(result, "", " ")
+
+			objectName := strings.ToLower(fmt.Sprintf("%s-%s-%s-%s.json.gz", insights.InputType, resource.Project, resource.Environment, resource.Service))
+			tempFilePath := fmt.Sprintf("/tmp/%s", objectName)
+			contentType := "application/gzip"
+
+			var buf bytes.Buffer
+			gz := gzip.NewWriter(&buf)
+			gz.Write(resultJson)
+			gz.Close()
+			err = ioutil.WriteFile(tempFilePath, buf.Bytes(), 0644)
+			if err != nil {
+				fmt.Errorf(err.Error())
+			}
+
+			info, err := minioClient.FPutObject(ctx, h.S3Config.Bucket, objectName, tempFilePath, minio.PutObjectOptions{ContentType: contentType})
+			if err != nil {
+				fmt.Errorf(err.Error())
+			}
+			log.Printf("Successfully uploaded %s of size %d\n", objectName, info.Size)
+
+			err = os.Remove(tempFilePath)
+			if err != nil {
+				fmt.Errorf(err.Error())
+			}
+		}
+	}
+
+	return nil
+}
+
+func (h *Messaging) pushFactsToLagoonApi(facts []lagoonclient.AddFactInput, resource ResourceDestination) error {
+	apiClient := graphql.NewClient(h.LagoonAPI.Endpoint, &http.Client{Transport: &authedTransport{wrapped: http.DefaultTransport, h: h}})
+
+	log.Println("--------------------")
+	log.Printf("Attempting to add %d facts...", len(facts))
+	log.Println("--------------------")
+
+	result, err := lagoonclient.AddFacts(context.TODO(), apiClient, facts)
+	if err != nil {
+		return err
+	}
+
+	log.Println(result)
+	return nil
+}
+
+func decodeString(encodedString string) (result interface{}, err error) {
+	// base64 decode it
+	base64Decoder := base64.NewDecoder(base64.StdEncoding, strings.NewReader(encodedString))
+	decodedGzipReader, err := gzip.NewReader(base64Decoder)
+	if err != nil {
+		return "", err
+	}
+
+	// Decode json from reader
+	var data interface{}
+	jsonDecoder := json.NewDecoder(decodedGzipReader)
+	err = jsonDecoder.Decode(&data)
+	if err != nil && err != io.EOF {
+		return "", err
+	}
+
+	return data, nil
+}
+
+func downloadSBOM(sbomURL string) (*http.Response, error) {
+	res, err := http.Get(sbomURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func processFactsFromSBOM(facts *[]cdx.Component, environmentId int, source string) []lagoonclient.AddFactInput {
+	var factsInput []lagoonclient.AddFactInput
+	if len(*facts) == 0 {
+		return factsInput
+	}
+
+	keyFacts, err := scanKeyFactsFile("./key_facts.txt")
```

Comment:
this file isn't currently in the image - so the handler fails here - we either copy it into the dockerfile, or mount it via a configMap